src/mlpack/methods/reinforcement_learning/environment/acrobat.hpp

```diff
@@ -0,0 +1,317 @@
+                          state.AngularVelocity1(),
+                          state.AngularVelocity2()};
+    double torque = Torque(action);
+    arma::colvec nextstate = Rk4(state_, torque);
+
+    nextState.Theta1() = Wrap(nextstate[0], -M_PI, M_PI);
+
+    nextState.Theta2() = Wrap(nextstate[1], -M_PI, M_PI);
+    nextState.AngularVelocity1() = Bound(nextstate[2],
+                                             -maxVel1 , maxVel1);
+    nextState.AngularVelocity2() = Bound(nextstate[3],
+                                             -maxVel2 , maxVel2);
+    return -1;
+  };
+  double Sample(const State& state, const Action& action) const
+  {
+    State nextState;
+    return Sample(state, action, nextState);
+  }
+  State InitialSample() const
+  {
+    return State((arma::randu<arma::colvec>(4) - 0.5) / 5.0);
+  }
+  /**
+   *
+   * @param state The current State
+   */        
+  bool IsTerminal(const State& state) const
+  {
+    return bool (-cos(state.Theta1())-cos(state.Theta1()
+      + state.Theta2()) > 1.0);
+  }
+  /**
+   * @param state Current State
+   * @param torque Torque Applied 
+   */
+  arma::colvec Dsdt(arma::colvec state,
+               const double torque) const
+  {
+    double m1 = linkMass1;
+    double m2 = linkMass2;
+    double l1 = linkLength1;
+    double l2 = linkLength2;
+    double lc1 = linkCom1;
+    double lc2 = linkCom2;
+    double I1 = linkMoi;
+    double I2 = linkMoi;
+    double g = gravity;
+    double a = torque;
+    double theta1 = state[0];
+    double theta2 = state[1];
+    double dtheta1 = state[2];
+    double dtheta2 = state[3];
+    double d1 = m1 * pow(lc1, 2) + m2 *
+              (pow(l1, 2) + pow(lc2, 2) + 2 * l1 * lc2 * cos(theta2))
+               + I1 + I2;
+    double d2 = m2 * (pow(lc2, 2) + l1 * lc2 * cos(theta2)) + I2;
+    double phi2 = m2 * lc2 * g * cos(theta1 + theta2 - M_PI / 2.);
+
+    double phi1 = - m2 * l1 * lc2 * pow(dtheta2, 2) * sin(theta2)
+      - 2 * m2 * l1 * lc2 * dtheta2 * dtheta1 * sin(theta2)
+      + (m1 * lc1 + m2 * l1) * g * cos(theta1 - M_PI / 2)
+      + phi2;
+
+    double ddtheta2 = (a + d2 / d1 * phi1 - m2 * l1 * lc2 * pow(dtheta1, 2) *
+     sin(theta2) - phi2) / (m2 * pow(lc2, 2) + I2 - pow(d2, 2) / d1);
+
+    double ddtheta1 = -(d2 * ddtheta2 + phi1) / d1;
+    arma::colvec returnValues = {dtheta1, dtheta2, ddtheta1, ddtheta2};
+    return returnValues;
+  };
+  /**
+   * @param value scalar value to wrap
+   * @param minimum minimum range of wrap
+   * @param maximum maximum range of wrap
+   */      
+  double Wrap(double value,
+              double minimum,
+              double maximum) const
+  {
+    double diff = maximum - minimum;
+    if (value > maximum) value = value - diff;
+    else if (value < minimum) value = value + diff;
+    return value;
+  };
+  /**
+    * @param value scalar value to bound
+    * @param minimum minimum range of bound
+    * @param maximum maximum range of bound
+    */     
+  double  Bound(double value,
```

Comment:
Not sure, an extra function is necessary here, since the actual implementation is quite simple and it's only used twice. Let me know what you think.

Reply:
I think it is needed here. There are chances that angle may go out of bound. To bring back we need a wrap function. Like 5pi/2 is ideally pi/2. If we not use the wrap feature state space will unnecessarily increase?

How do you see this?

Reply:
Agreed, but do you think we need an extra function, or could we directly use ``std::min(std::max(value, minimum), maximum);``.